Adam

Atom-Cunningham opened the pull request:
Official course information is being read into the postgresql database from the go server via gorm

Atom-Cunningham committed:
Updated models

Updated the gorm models to match the database schema. Added some basic csv/json processing (untested)

Atom-Cunningham committed:
added read/write operations for the database

duplication of effort with Ramya?

Atom-Cunningham committed:
Small writes to db

Atom-Cunningham committed:
Official list of courses added to initdb

Atom-Cunningham committed:
commented out setup function

Atom-Cunningham committed:
Frontend refactor

Preparing to make axios requests to gin server. Rebuild may be required

Atom-Cunningham committed:
Rebuilt

Atom-Cunningham committed:
api call working

in router.go, handlers assigned to api group "/courses".
http://localhost:3000/api/courses/ working

krishnaprashanth246 commented on backend/controllers/courses.go:
```diff
@@ -0,0 +1,86 @@
+	var courses []models.Course
+
+	// Fetch courses from the database
+	result := models.DB.Find(&courses)
```
This is gorm code, all the code related to db read or write and gorm related code should be inside models/* files. The controller functions should call those functions instead of directly invoking gorm code. Applicable for every controller function(AKA handlers).

Atom-Cunningham committed:
End-to-end pipeline complete

Atom-Cunningham committed:
updated professor spreadsheet

Atom-Cunningham committed:
imported new frontend

Atom-Cunningham committed:
debugging frontend

Atom-Cunningham committed:
.

Atom-Cunningham committed:
New Frontend

Exposing the matchmaking service

Atom-Cunningham committed:
Finished gorm

Polished the database interface. Tested writing via gorm models into every table.

Atom-Cunningham committed:
dont auto update db
